packages/manager/core/request-tagger/__tests__/request-tagger.spec.ts

```diff
@@ -0,0 +1,319 @@
+  });
+
+  it('returns a version header when defined', () => {
+    const applicationVersion = 'version-1234';
+    defineApplicationVersion(applicationVersion);
+    const { [Header.VERSION]: versionHeader } = getHeaders(
+      '/engine/apiv6/endpoint',
+    );
+    expect(versionHeader).toEqual(applicationVersion);
+  });
+});
+
+describe('Headers override', () => {
+  afterEach(() => {
+    const overrides = getHeadersOverride();
+    Object.keys(overrides).forEach((pattern) => {
+      removeHeadersOverride(pattern);
+    });
+  });
+
+  it('overrides headers for specific calls', () => {
+    const defaultApplicationVersion = 'default-version';
+    const defaultCurrentPage = 'default-page';
+    defineApplicationVersion(defaultApplicationVersion);
+    defineCurrentPage(defaultCurrentPage);
+
+    const secretVersion = 'secret-version';
+    const secretPage = 'secret-page';
+    addHeadersOverride('.*/secret$', {
+      [Header.VERSION]: secretVersion,
+      [Header.PAGE]: secretPage,
+    });
+
+    const {
+      [Header.VERSION]: secretVersionHeader,
+      [Header.PAGE]: secretPageHeader,
+    } = getHeaders('/engine/apiv6/endpoint/secret');
+    expect(secretVersionHeader).toEqual(secretVersion);
+    expect(secretPageHeader).toEqual(secretPage);
+
+    const {
+      [Header.VERSION]: notSecretVersionHeader,
+      [Header.PAGE]: notSecretPageHeader,
+    } = getHeaders('/engine/apiv6/endpoint/not-secret');
+    expect(notSecretVersionHeader).toEqual(defaultApplicationVersion);
+    expect(notSecretPageHeader).toEqual(defaultCurrentPage);
+  });
+
+  it('allows to remove overriden headers', () => {
+    const defaultApplicationVersion = 'default-version';
+    const defaultCurrentPage = 'default-page';
+    defineApplicationVersion(defaultApplicationVersion);
+    defineCurrentPage(defaultCurrentPage);
+
+    const secretVersion = 'secret-version';
+    const secretPage = 'secret-page';
+    const secretPattern = '.*/secret$';
+    addHeadersOverride(secretPattern, {
+      [Header.VERSION]: secretVersion,
+      [Header.PAGE]: secretPage,
+    });
+
+    const commonPage = 'common';
+    addHeadersOverride('.*/common$', {
+      [Header.PAGE]: commonPage,
+    });
+
+    const {
+      [Header.VERSION]: secretVersionHeader,
+      [Header.PAGE]: secretPageHeader,
+    } = getHeaders('/engine/apiv6/endpoint/secret');
+    expect(secretVersionHeader).toEqual(secretVersion);
+    expect(secretPageHeader).toEqual(secretPage);
+
+    const { [Header.PAGE]: commonPageHeader } = getHeaders(
+      '/engine/apiv6/endpoint/common',
+    );
+    expect(commonPageHeader).toEqual(commonPage);
+
+    const {
+      [Header.VERSION]: notSecretVersionHeader,
+      [Header.PAGE]: notSecretPageHeader,
+    } = getHeaders('/engine/apiv6/endpoint/not-secret');
+    expect(notSecretVersionHeader).toEqual(defaultApplicationVersion);
+    expect(notSecretPageHeader).toEqual(defaultCurrentPage);
```

Comment:
Same here

Reply:
Should be ok now, we check in this test that overrides don't apply on URL that doesn't match the override pattern.